### PR TITLE
003 information tooltips

### PR DESCRIPTION
**Pull request by aknipey**
Poor quality content of the actual information. Need to improve before actually merged in. Please help, especially on the coin-toss feature (couldnt be bothered to figure out the maths its easier to ask you)
Screenshot from 2023-09-08 00-01-30

### CODE REVIEW

**Comment by patrickemoore** [comment]
We will iterate on this. For now have a look at https://mui.com/material-ui/react-floating-action-button/ and replace the button

File: frontend/src/components/OutcomeGraph.tsx [hunk 1]
```diff
@@ -91,48 +105,111 @@ export function OutcomeGraph(props: {
...
         mode: "lines",
         marker: { color: "green" },
         showlegend: false
       }
     });
-  }, [extendedSimulation]);
+  }, [extendedSimulation, normalised, percentageChangeAxis]);
+
+
+  const GRAPH_LAYOUT = useMemo(() => {
```
Comment (patrickemoore) [inline frontend/src/components/OutcomeGraph.tsx:151]: THIS_NAMING_SCHEME is only for global scope constants, useThisOneInstead for in-component constants

File: frontend/src/components/OutcomeGraph.tsx [hunk 2]
```diff
@@ -91,48 +105,111 @@ export function OutcomeGraph(props: {
...
+      }
+    } else if (percentageChangeAxis && !normalised) {
+      y = {
+        title: "Portfolio Value - %"
+      } 
+    } else {
+      y = {
+        title: "Portfolio Value - $"
+      }
+    }
```
Comment (patrickemoore) [inline frontend/src/components/OutcomeGraph.tsx:169]: construct the title in 3 parts, the "Portfolio Value", "Change" and "$" or "%". Also factor out the y = {}

File: frontend/src/components/OutcomeGraph.tsx [hunk 3]
```diff
@@ -91,48 +105,111 @@ export function OutcomeGraph(props: {
...
+    return({
+    responsive: true,
+    title: "Portfolio Values",
+    autosize: true,
+    width: 1200,
+    height: 700,
+    xaxis: {
+      title: "Months",
+    },
```
Comment (patrickemoore) [inline frontend/src/components/OutcomeGraph.tsx:178]: Define this as global constant and spread it

File: frontend/src/components/OutcomeGraph.tsx [hunk 4]
```diff
@@ -91,48 +105,111 @@ export function OutcomeGraph(props: {
...
+          if (normalised && percentageChangeAxis) {
+            return portfolioState.realised / outcome[0].realised * 100 - 100;
+          } else if (normalised && !percentageChangeAxis) {
+            return portfolioState.realised - outcome[0].realised;
+          } else if (!normalised && percentageChangeAxis) {
+            return portfolioState.realised / outcome[0].realised * 100;
+          } else {
+            return portfolioState.realised;
+          }
+        }),
```
Comment (patrickemoore) [inline frontend/src/components/OutcomeGraph.tsx:196]: Extract into additional functions

**Review by patrickemoore — CHANGES_REQUESTED** [review]

File: frontend/src/components/OutcomeGraph.tsx (hunk 4 above)
Comment (aknipey) [reply frontend/src/components/OutcomeGraph.tsx:196]: already done in 002

File: frontend/src/components/OutcomeGraph.tsx (hunk 1 above)
Comment (aknipey) [reply frontend/src/components/OutcomeGraph.tsx:151]: i may have copied that scheme in more spots bc this particular one was already in the code. Will change

**Review by patrickemoore — APPROVED** [review]

**Review by patrickemoore — APPROVED** [review]

File: frontend/src/components/ParameterInput.tsx [hunk 5]
```diff
@@ -309,22 +343,58 @@ export function ParameterInput(props: {
...
+                  </Typography>
+                  <Typography variant="body1">
+                    <strong>Complex Die:</strong> A die with 10 sides, of which the return given on each side is determined by the multiplier input into each box
+                  </Typography>
+                  <Typography variant="body1">
+                    <strong>Coin Toss:</strong> When enables, if a win side is rolled on the simple die, there is a chance to increase returns by the winMultiplier. The coin is kept flipping until you lose. This feature is used to balance out the tail end of the distribution
+                  </Typography>
+                </DialogContent>
+              </Dialog>
+            </Stack>
```
Comment (patrickemoore) [inline frontend/src/components/ParameterInput.tsx:397]: This will have to be improved

**Review by patrickemoore — APPROVED** [review]
lgtm, see inline

**Pull request merged**